src/main/java/com/uber/profiling/Arguments.java

```diff
@@ -152,71 +156,114 @@ public void updateArguments(Map<String, List<String>> parsedArgs) {
+                }
+                String className = classMethodName.substring(0, index);
+                String methodName = str.substring(index + 1, classMethodName.length());
 
-            ClassMethodArgument classMethodArgument = new ClassMethodArgument(className, methodName, argumentIndex);
-            argumentProfiling.add(classMethodArgument);
-            logger.info("Got argument value for argumentProfiling: " + classMethodArgument);
+                ClassMethodArgument classMethodArgument = new ClassMethodArgument(className, methodName, argumentIndex);
+                argumentProfiling.add(classMethodArgument);
+                logger.info("Got argument value for argumentProfiling: " + classMethodArgument);
+            }
         }
 
-        brokerList = getArgumentSingleValue(parsedArgs, ARG_BROKER_LIST);
-        logger.info("Got argument value for brokerList: " + brokerList);
+        argValue = getArgumentSingleValue(parsedArgs, ARG_BROKER_LIST);
+        if (needToUpdateArg(argValue)) {
+            brokerList = argValue;
+            logger.info("Got argument value for brokerList: " + brokerList);
+        }
 
         argValue = getArgumentSingleValue(parsedArgs, ARG_SYNC_MODE);
-        if (argValue != null && !argValue.isEmpty()) {
+        if (needToUpdateArg(argValue)) {
             syncMode = Boolean.parseBoolean(argValue);
             logger.info("Got argument value for syncMode: " + syncMode);
         }
 
-        topicPrefix = getArgumentSingleValue(parsedArgs, ARG_TOPIC_PREFIX);
-        logger.info("Got argument value for topicPrefix: " + topicPrefix);
+        argValue = getArgumentSingleValue(parsedArgs, ARG_TOPIC_PREFIX);
+        if (needToUpdateArg(argValue)) {
+            topicPrefix = argValue;
+            logger.info("Got argument value for topicPrefix: " + topicPrefix);
+        }
 
-        outputDir = getArgumentSingleValue(parsedArgs, ARG_OUTPUT_DIR);
-        logger.info("Got argument value for outputDir: " + outputDir);
+        argValue = getArgumentSingleValue(parsedArgs, ARG_OUTPUT_DIR);
+        if (needToUpdateArg(argValue)) {
+            outputDir = argValue;
+            logger.info("Got argument value for outputDir: " + outputDir);
+        }
 
         argValue = getArgumentSingleValue(parsedArgs, ARG_IO_PROFILING);
-        if (argValue != null && !argValue.isEmpty()) {
+        if (needToUpdateArg(argValue)) {
             ioProfiling = Boolean.parseBoolean(argValue);
             logger.info("Got argument value for ioProfiling: " + ioProfiling);
         }
     }
     
+    public void processConfigProvider() {
+        try {
+            ConfigProvider configProvider = getConfigProvider();
+            if (configProvider != null) {
+                Map<String, Map<String, List<String>>> extraConfig = configProvider.getConfig();
+
+                Map<String, List<String>> rootConfig = extraConfig.get("");
```

Comment:
would you add some comments about the contract here, why "" represents "root config"?

Reply:
Good suggestion. Will add comment.